### PR TITLE
Update build flags for SYCL, Kokkos, and OpenMP, tracking newest versions of each compiler

### PR DESCRIPTION
**Pull request by tom91136**
This supersedes #69

### CODE REVIEW

**Comment by tomdeakin** [comment]
Looks good overall.

Does this work such that the `SYCLStream.sycl` target is *not built* when DPCPP and hipSYCL compilers are selected?

Please can you add the DPCPP NVIDIA target with the flags shown in #69?

**Comment by tom91136** [comment]
@tomdeakin this is ready for review+merge

File: SYCL.make [hunk 1]
```diff
@@ -1,17 +1,95 @@
...
+
+SYCL_SYCLFLAGS = $(SYCL_$(COMPILER)_SYCLFLAGS_$(TARGET))
+SYCL_SYCLCXX = $(SYCL_$(COMPILER)_SYCLCXX)
+
+SYCL_FLAGS = $(SYCL_$(COMPILER)_FLAGS)
+SYCL_LINK_FLAGS = $(SYCL_$(COMPILER)_LINK_FLAGS)
+SYCL_INCLUDE = $(SYCL_$(COMPILER)_INCLUDE)
+SYCL_CXX = $(SYCL_$(COMPILER)_CXX)
 
 sycl-stream: main.o SYCLStream.o SYCLStream.sycl
```
[the remark this answers is not in the document]
Comment (tom91136) [reply SYCL.make:82]: Yep , this made it compile much faster too.
Turns out there was also a bug where the content of  `SYCL_COMPUTECPP_SYCLFLAGS` wasn't used.

File: Kokkos.make [hunk 2]
```diff
@@ -56,6 +59,9 @@ Set TARGET to change to offload device. Defaulting to CPU.
 Available targets are:
   CPU (default)
   GPU
+
+  Note: Use CPU or leave default if compiling with HIPCC
```
[the remark this answers is not in the document]
Comment (tom91136) [reply Kokkos.make]: Done

File: OpenMP.make [hunk 3]
```diff
@@ -16,7 +16,7 @@ ifndef TARGET
 define target_help
 Set TARGET to change device (defaulting to CPU).
 Available targets are:
-  CPU NVIDIA AMD
+  CPU NVIDIA AMD INTEL_GT
```
[the remark this answers is not in the document]
Comment (tom91136) [reply OpenMP.make]: Done

**Pull request merged**